Dockerfile

```diff
@@ -0,0 +1,107 @@
+    update-alternatives --install /usr/bin/x86_64-linux-gnu-cpp x86_64-linux-gnu-cpp /usr/bin/g++-$GNUCOMPILER 100 && \
+    update-alternatives --install /usr/bin/gfortran gfortran /usr/bin/gfortran-$GNUCOMPILER 100 && \
+    update-alternatives --install /usr/bin/x86_64-linux-gnu-gfortran x86_64-linux-gnu-gfortran /usr/bin/gfortran-$GNUCOMPILER 100 && \
+    update-alternatives --install /usr/bin/python python /usr/bin/python3.6 100
+
+RUN mkdir -p /optsoft/bin && \
+    mkdir -p /optsoft/lib && \
+    mkdir -p /optsoft/src && \
+    mkdir -p /optsoft/include
+
+ENV PATH $PATH:/optsoft/bin
+ENV LD_LIBARY_PATH $LD_LIBRARY_PATH:/optsoft/lib
+ENV C_INCLUDE_PATH $C_INCLUDE_PATH:/optsoft/include
+ENV CPLUS_INCLUDE_PATH $C_INCLUDE_PATH:/optsoft/include
+
+# ---------- Build RNXCMP from source into /optsoft ---------- 
+RUN mkdir /src
+ADD RNXCMP_4.0.5_src.tar.gz /src/
+WORKDIR /src/RNXCMP_4.0.5_src/source
+RUN gcc-$GNUCOMPILER -ansi -O2 -static rnx2crx.c -o /optsoft/bin/RNX2CRX && \
+    gcc-$GNUCOMPILER -ansi -O2 -static crx2rnx.c -o /optsoft/bin/CRX2RNX
+
+# ---------- COMPILE ALBUS ----------
+# Step 1 setup directory structure and move source code into container
+RUN mkdir /src/ALBUS
+ENV ALBUSPATH /src/ALBUS
+
+ADD IMF24_request.png $ALBUSPATH
+ADD build_rnx_crx $ALBUSPATH
+ADD dates $ALBUSPATH
+ADD definitions $ALBUSPATH
+ADD fast_orbital_data $ALBUSPATH
+ADD INSTALL $ALBUSPATH
+ADD Makefile $ALBUSPATH
+ADD iri.update $ALBUSPATH
+ADD kill_python $ALBUSPATH
+ADD LICENSE $ALBUSPATH
+ADD README.md $ALBUSPATH
+ADD remove_remainder $ALBUSPATH
+ADD test_iri.f $ALBUSPATH
+ADD UPDATING_SPACE_WEATHER $ALBUSPATH
+ADD bin $ALBUSPATH/bin
+ADD C++ $ALBUSPATH/C++
+ADD examples $ALBUSPATH/examples
+ADD FORTRAN $ALBUSPATH/FORTRAN
+ADD include $ALBUSPATH/include
+ADD IRI_2012 $ALBUSPATH/IRI_2012
+ADD libdata $ALBUSPATH/libdata
+ADD Python $ALBUSPATH/Python
+ADD python_scripts $ALBUSPATH/python_scripts
+ADD share $ALBUSPATH/share
+
+# Step 2 apply necessary patches to build system
+RUN sed -i '15s/.*/export INSTALLDIR = '$(echo ${ALBUSPATH} | sed 's/\//\\\//g')'/' $ALBUSPATH/Makefile
+#### Ubuntu 18.04 ships Python 3.6 LTS not 3.8 as it is defined in the build system
+RUN sed -i '19s/.*/export CURRENT_PYTHON = python3.6/' $ALBUSPATH/Makefile
+RUN sed -i '22s/.*/export PYTHONINCLUDEDIR = \/usr\/include\/python3.6/' $ALBUSPATH/Makefile
+#### Build with specified GNU toolchain
+RUN sed -i '39s/.*/export CC = gcc-'"${GNUCOMPILER}"'/' $ALBUSPATH/Makefile
+RUN sed -i '40s/.*/export F77 = gfortran-'"${GNUCOMPILER}"' --std=legacy/' $ALBUSPATH/Makefile
+RUN sed -i '43s/.*/export F77_RECL_UNIT = bytes/' $ALBUSPATH/Makefile
+RUN sed -i '45s/.*/export C++ = g++-'"${GNUCOMPILER}"'/' $ALBUSPATH/Makefile
+RUN sed -i '47s/.*/export CPP = cpp-'"${GNUCOMPILER}"' -P/' $ALBUSPATH/Makefile
+RUN sed -i '10s/.*/CFLAGS += -I$(PYTHONINCLUDEDIR) -I$(INSTALLDIR)\/include -DINSTALLDIR=''$(INSTALLDIR)''/' $ALBUSPATH/C++/mim/test/PIMrunner/Makefile
```

Comment:
Have to pass -DINSTALLDIR to get the variable passed through gcc. Not sure how this worked previously